client/image_list_test.go

```diff
@@ -163,62 +165,37 @@ func TestImageListApiBefore125(t *testing.T) {
 // Checks if shared-size query parameter is set/not being set correctly
 // for /images/json.
 func TestImageListWithSharedSize(t *testing.T) {
-	// performs ImageList call and returns the sent url query
-	getImageListAndGetQuery := func(version string, options types.ImageListOptions) (query url.Values, err error) {
-		client := &Client{
-			client: newMockClient(func(req *http.Request) (*http.Response, error) {
-				query = req.URL.Query()
-
-				return &http.Response{
-					StatusCode: http.StatusOK,
-					Body:       io.NopCloser(strings.NewReader("[]")),
-				}, nil
-			}),
-			version: version,
-		}
-
-		_, err = client.ImageList(context.Background(), options)
-		return query, err
-	}
-
+	t.Parallel()
 	const sharedSize = "shared-size"
-
-	// Check if the shared-size query parameter is unset
-	for _, tC := range []struct {
-		name    string
-		version string
-		options types.ImageListOptions
+	for _, tc := range []struct {
+		name       string
+		version    string
+		options    types.ImageListOptions
+		sharedSize string // value the shared-size query param should be set to, or empty if it should not be set.
 	}{
 		{name: "after 1.42, if not requested", version: "1.42", options: types.ImageListOptions{SharedSize: false}},
```

Comment:
I guess for this one we could remove the options altogether to be more clear "nothing is set". Let me update that as well.